ft-transcendence/transcendence
# 🚀 Shlu/feature/complete chat

yay chat is done! now we need some hard tests 🎉

to test:
  0. clear the database if it was built already, cause the schema is updated!
1. sign up several accounts so you can talk to yourself
    > before your sign up any users, if your browser is still in logged in status there will be error it's normal
2. create different types of chat (public/private/protected with password)
3. assign users as admin/take back admin right/change the channel setting, if you're owner
4. kick out/mute users, if you're admin
5. invite users to channel, if you're at least a member
6. search for user to DM/public chat/my chat
7. user will be able to see private channel only if they are invited, 
    if the channel is protected by password user still need to be checked by password
8. join/leave/block channels
9. send/unsend msgs
10. log out and back in to see if data is correct

some must do rely on other parts that we'll deal with it later:
1. display online status
2. fetch searching suggestion to all the users while new user sign up 
3. add friend/block user
4. invite to a game
5. link to user profile
6. display avatar
... and more i haven't think about ...

those we maybe can do:
1. last msg sent time in preview
2. msgs sent time in chatroom
3. edit msg
4. unread count show up on preview

hope you like it 🥰
